src/ResourceManagement/Logic/Logic.Tests/InMemoryTests/IntegrationAccountSessionInMemoryTests.cs

```diff
@@ -0,0 +1,316 @@
+            Assert.Throws<CloudException>(() => client.Sessions.ListByIntegrationAccountsNext(nextPageLink: Constants.NextPageLink));
+        }
+
+        [Fact]
+        public void IntegrationAccountSession_ListByResourceGroupNext_Success()
+        {
+            var handler = new RecordedDelegatingHandler();
+            var client = this.CreateIntegrationAccountClient(handler);
+
+            handler.Response = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = this.SessionList
+            };
+
+            var result = client.Sessions.ListByIntegrationAccountsNext(nextPageLink: Constants.NextPageLink);
+
+            handler.Request.ValidateAuthorizationHeader();
+            handler.Request.ValidateMethod(HttpMethod.Get);
+
+            this.ValidateSessionList(result);
+        }
+
+        [Fact]
+        public void IntegrationAccountSession_CreateOrUpdate_Exception()
+        {
+            var handler = new RecordedDelegatingHandler();
+            var client = this.CreateIntegrationAccountClient(handler);
+
+            handler.Response = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.InternalServerError,
+                Content = new StringContent(string.Empty)
+            };
+
+            Assert.Throws<ValidationException>(() => client.Sessions.CreateOrUpdate(
+                resourceGroupName: null,
+                integrationAccountName: "IntegrationAccountName",
+                sessionName: "SessionName",
+                session: new IntegrationAccountSession()));
+            Assert.Throws<ValidationException>(() => client.Sessions.CreateOrUpdate(
+                resourceGroupName: ResourceGroupName,
+                integrationAccountName: null,
+                sessionName: "SessionName",
+                session: new IntegrationAccountSession()));
+            Assert.Throws<ValidationException>(() => client.Sessions.CreateOrUpdate(
+                resourceGroupName: ResourceGroupName,
+                integrationAccountName: "IntegrationAccountName",
+                sessionName: null,
+                session: new IntegrationAccountSession()));
+            Assert.Throws<ValidationException>(() => client.Sessions.CreateOrUpdate(
+                resourceGroupName: ResourceGroupName,
+                integrationAccountName: "IntegrationAccountName",
+                sessionName: "SessionName",
+                session: null));
+            Assert.Throws<CloudException>(() => client.Sessions.CreateOrUpdate(
+                resourceGroupName: ResourceGroupName,
+                integrationAccountName: "IntegrationAccountName",
+                sessionName: "SessionName",
+                session: new IntegrationAccountSession()));
+        }
+
+        [Fact]
+        public void IntegrationAccountSession_CreateOrUpdate_OK()
+        {
+            var handler = new RecordedDelegatingHandler();
```

Comment:
I am not sure I understand the intent of the positive tests in this test class - it looks like you are using a canned response to verify the authorization header and the request method.  Are these likely to change?  Why couldn't these tests be more easily performed with a single recorded test?

The neagtive tests make a lot of sense, just not sure I understand the reasoning behind the positive tests in this class